### PR TITLE
Added check for GNUtls in FIPS140 mode

### PR DESCRIPTION
**Pull request by dcermak**
[CI:TOXENVS] fips

CI PR for #678

### CODE REVIEW

File: tests/test_fips.py [hunk 1]
```diff
@@ -189,14 +190,19 @@ def test_gnutls_binary(container_per_test: ContainerData) -> None:
...
         "gcc -Og -g3 fips-test-gnutls.c -Wall -Wextra -Wpedantic -lgnutls -o fips-test-gnutls && "
         "mv fips-test-gnutls /bin/fips-test-gnutls"
     )
 
-    assert "NOT" not in container_per_test.connection.check_output(
-        "gnutls-cli --fips140-mode"
+    c.check_output("gnutls-cli --fips140-mode")
+
+    assert search(
+        r"library is in FIPS140(-3|-2|) mode",
```
Comment (dirkmueller) [inline tests/test_fips.py:204]: do we really need to support the empty match, `|)` here? which distribution produces that output?

Comment (dcermak) [reply tests/test_fips.py:202]: Please add this to #678 instead

**Review by ricardobranco777 — APPROVED** [review]

File: tests/test_fips.py (hunk 1 above)
Comment (dcermak) [reply tests/test_fips.py:202]: @pablo-herranz ^^^

Comment (pablo-herranz) [reply tests/test_fips.py:202]: Yes, we do as the output message might be:
- library is in FIPS140-3 mode
- library is in FIPS140-2 mode
- library is in FIPS140 mode

**Review by grisu48 — APPROVED** [review]

**Pull request merged**